src/lread.c

```diff
@@ -4009,6 +4009,20 @@ check_obarray (Lisp_Object obarray)
   return obarray;
 }
 
+/* Like check_obarray, but for the global Vobarray. */
```

Comment:
Ouch, that's odd. Let's open an issue and fix that, but I don't think it should hold up this PR.